compose-lint-checks/src/main/java/slack/lint/compose/ViewModelInjectionDetector.kt

```diff
@@ -57,7 +71,10 @@ class ViewModelInjectionDetector : ComposableFunctionDetector(), SourceCodeScann
       .flatMap { property ->
         property
           .findDirectChildrenByClass<KtCallExpression>()
-          .filter { KnownViewModelFactories.contains(it.calleeExpression?.text) }
+          .filter {
+            KnownViewModelFactories.contains(it.calleeExpression?.text) ||
+              allowedNames.value.contains(it.calleeExpression?.text)
```

Comment:
Let's merge the known and allowed factories into one set to avoid double checks. Let's also use the `in` operator